leaderboardsgg/leaderboard-site
# [589] Fix issue with password confirmation in sign-up

## What

- Fixes Password & Password confirmation logic in `Sign Up` card
- Also changes it so that there isn't 3 password unmask buttons in Password Sign Up

## Link to Issue

Closes https://github.com/leaderboardsgg/leaderboard-site/issues/589

## Acceptance

### Steps for testing

1. Open the page
2. Click Sign-up
3. Enter a valid Email and Username
4. Click the un-hide password icon
5. Type in `test123` in password
6. Type in `test123` in password confirmation
7. verify that the clickable cursor appears over the sign up button
8. go to password confirm and backspace the 3
9. verify that the clickable cursor no longer appears over the sign up button
10. fix the passwords to match again and verify you can sign up again
11. go to the password field (instead of confirmation) and backspace
12. verify there is an error that shows & you can no longer sign up
13. fix the passwords again and finally verify you can sign up

### Images
https://github.com/leaderboardsgg/leaderboard-site/assets/11199709/f72a87c2-9edf-4f7e-ac4c-d62c2b199898
